### PR TITLE
RavenDB-8375: MinDataRateDisable + MinDataRateBytesPerSecond + MinDat…

### PR DESCRIPTION
**Pull request by aviviadi**
…aRateGracePeriod + MaxRequestBufferSizeInKb Configuration options

### CODE REVIEW

File: src/Raven.Server/Config/Categories/CoreConfiguration.cs [hunk 1]
```diff
@@ -52,6 +52,27 @@ public class CoreConfiguration : ConfigurationCategory
...
+        [Description("Set Kestrel's allowed request and reponse grace in second. This option should configured together with MinDataRateBytesPerSecond. If MinDataRateDisable set to true - this option will be ignored")]
+        [DefaultValue(0)]
+        [ConfigurationEntry("MinDataRateGracePeriod")]
+        public int MinDataRateGracePeriod { get; set; }
+
+        [Description("Set Kestrel's MaxRequestBufferSize. Value is in KB. If set to 0 - then Kestrel's defaults are applied (1MB)")]
+        [DefaultValue(0)]
+        [ConfigurationEntry("MaxRequestBufferSizeInKb")]
+        public int MaxRequestBufferSizeInKb { get; set; }
+
```
Comment (arekpalinski) [inline src/Raven.Server/Config/Categories/CoreConfiguration.cs:74]: All of them seem to be nice candidates to be put into a separate category `Http`? Then we can name them `Http.MaxRequestBufferSizeInKb ` etc

File: src/Raven.Server/Config/Categories/CoreConfiguration.cs [hunk 2]
```diff
@@ -52,6 +52,27 @@ public class CoreConfiguration : ConfigurationCategory
...
 
+        [Description("Disable Kestrel's minimum required data rate with grace time enforcement. If set to true this option overrides both MinDataRateBytesPerSecond and MinDataRateGracePeriod settings")]
+        [DefaultValue(false)]
+        [ConfigurationEntry("MinDataRateDisable")]
+        public bool MinDataRateDisable { get; set; }
+
+        [Description("Set Kestrel's minimum required data rate in bytes per second. This option should configured together with MinDataRateGracePeriod. If MinDataRateDisable set to true - this option will be ignored")]
+        [DefaultValue(0)]
+        [ConfigurationEntry("MinDataRateBytesPerSecond")]
+        public int MinDataRateBytesPerSecond { get; set; }
```
Comment (arekpalinski) [inline src/Raven.Server/Config/Categories/CoreConfiguration.cs:63]: Let's use `TimeSetting`, then you can avoid `TimeSpan.FromSeconds()`. Also for consistency with other names let's name it `MinDataRateBytesPerSec`

File: src/Raven.Server/Config/Categories/CoreConfiguration.cs [hunk 3]
```diff
@@ -52,6 +52,27 @@ public class CoreConfiguration : ConfigurationCategory
...
+
+        [Description("Set Kestrel's allowed request and reponse grace in second. This option should configured together with MinDataRateBytesPerSecond. If MinDataRateDisable set to true - this option will be ignored")]
+        [DefaultValue(0)]
+        [ConfigurationEntry("MinDataRateGracePeriod")]
+        public int MinDataRateGracePeriod { get; set; }
+
+        [Description("Set Kestrel's MaxRequestBufferSize. Value is in KB. If set to 0 - then Kestrel's defaults are applied (1MB)")]
+        [DefaultValue(0)]
+        [ConfigurationEntry("MaxRequestBufferSizeInKb")]
+        public int MaxRequestBufferSizeInKb { get; set; }
```
Comment (arekpalinski) [inline src/Raven.Server/Config/Categories/CoreConfiguration.cs:73]: We've got `Size` for dealing with such settings

File: src/Raven.Server/Config/Categories/CoreConfiguration.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Config/Categories/CoreConfiguration.cs:74]: Yes, that would be a good idea.

**Comment by ayende** [comment]
Please PR again with the fixes

**Pull request closed without merging**